### DISCUSSION TITLE
Redesign: Switchboard should be sexier than gnome-control-center

### DISCUSSION THREAD

**Discussion by quequotion** [elementary/settings#136] [Ideas]
I wouldn't be the first to point out that switchboard's layout is falling behind. I don't think it's even the first time I've brought this up. I think this app has been neglected while other parts of the desktop were polished; Switchboard should be polished too.

This could have a wide scope, so I am expecting a discussion and no rush to a conclusion.

First, let's take a look at the competition. GNOME redesigned gnome-control-center [a few years ago](https://blogs.gnome.org/aday/2016/01/13/a-settings-design-update/), and it looks _great_.

Switchboard still looks like [this](https://github.com/elementary/switchboard#switchboard).

- I've never been happy with the layout; the icons are more far apart than necessary, and also too small for their texts. I'm sure I brought this up years ago, when it still looked worse than old gnome-control-center that used the same concept but had better spacing and icon:text size ratio (maybe it was on google+?)

- We could do the same thing gnome has, and convert this layout to a sidebar with a panel for each plug (like switchboard-plug-elementary-tweaks); but maybe we can come up with something more clever, or a more clever spin on a similar design.

**Comment by danirabbit** [comment]
I think the first thing to ask is "what problems would a new design solve"? It sounds like you think the icons should be bigger, but moving to a sidebar design wouldn't have that result.

What's really most important in Switchboard is that users find the setting they're looking for.

I think what's probably going to be the biggest driver there is maybe doing some kind of card sorting or surveying to see if we're categorizing settings in a way that facilitates discovery. This would be the biggest driver in determining if a layout change is necessary. If it turns out that users expect many more top level categories then a grid starts to break down. But if it turns out that users expect fewer top level categories then it makes more sense to stick to the current design.

It seems that the GNOME Control Center design was driven by the desire to have more top level categories and a need for a responsive design that works well on phones. Since we don't have a need to make Switchboard work on phone-sized displays, I think we need to see some kind of reasoning about the tangible benefits a change in design would bring for discovery.

I'm not sure "it's been like that for a while" is a compelling argument for keeping things the same, but it isn't a very compelling argument for changing them either

**Reply by danirabbit** [reply]
Revisiting this since we are now looking at responsive. Some of the recent design experiments:

![Screenshot from 2024-01-26 11 46 12](https://github.com/elementary/switchboard/assets/7277719/60118f51-eb16-4c76-bf79-e97440f79781)
![Screenshot from 2024-01-26 11 33 58](https://github.com/elementary/switchboard/assets/7277719/dab93d81-c81e-4864-80e0-20597bd8eba5)
![Screenshot from 2024-01-26 11 33 43](https://github.com/elementary/switchboard/assets/7277719/cd6cbfbd-f492-443a-8849-844299ed28f4)

![Screenshot from 2024-01-26 09 14 13](https://github.com/elementary/switchboard/assets/7277719/49327dc7-0f02-4c8e-b971-828cf3dbe66e)
![Screenshot from 2024-01-26 09 12 26](https://github.com/elementary/switchboard/assets/7277719/f6205948-b218-4e5e-94f6-f97b9dc24ab7)

**Comment by quequotion** [comment]
>What's really most important in Switchboard is that users find the setting they're looking for.

No argument with that. My argument is that we should both maximize its usability _and_ give it good aesthetics. I do not think these are mutually exclusive. We do need some mockups; I'll try to come up with some over the next few days. 

>It sounds like you think the icons should be bigger, but moving to a sidebar design wouldn't have that result.

My point was to illustrate shortcomings in the current aesthetics, not suggest a course of action. A sidebar design is by no means the only option.

>"what problems would a new design solve"?

There was [this (annoying, somewhat offensive, mostly FUD) video](https://www.reddit.com/r/elementaryos/comments/eo0qtk/some_ux_problems_in_pantheon_desktop/?utm_medium=android_app&utm_source=share) posted on the subreddit yesterday. 

Edit: Should have warned you, around 4:18 this video mocks you and Cassidy personally (as Penn and Teller..) and I am sorry to expose you to such unwarranted negativity. 

There are actually two or three points made worth hearing, but more importantly observing the problems the user has makes it clear they simply don't have much experience with elementary OS--which is a problem itself (keeping in mind "the user is drunk": the point is to be intuitive, so users are not challenged to _figure out_ how things work or where they are, yes?)

Particular to swichboard was that a settings panel's content was aligned to the top left corner, while a button was aligned to the lower right corner, sending them miles apart if the window size increased (actually only a few hundred px, but it does look bad and people could get flustered and give up).

You could argue it's just that one plug that needs redesign (networking), but I don't think it's a bad idea to consider something more comprehensive (all the plugs should have comfortable, functional, high quality aesthetics, and so should switchboard itself).

Edit: Already had an idea. Both problems would be mitigated if switchboard and its plugs scaled (and distributed) their content according to window size; still need to address layout issues (and avoid scaling to rediculous sizes), but scaling would reduce the unnecessarily wide open spaces.

**Comment by quequotion** [comment]
Here's switchboard in it's [inital state](https://i.postimg.cc/6QVQg5tc/switchboard-elementarydark-initialstate.png) (elementary:dark theme).

Here's an [infographic](https://i.postimg.cc/yxnJ36kn/switchboard-elementarydark-issues.png) to illustrate some of the shortcomings I see with the current design of switchboard itself.

Because the icons are so far apart, there ends up being a lot of emptiness in the window. Of course this depends on the number of plugs one has installed, but in any case the distance between icons is larger than necessary. The text being wider than the icons is not a good look; I'd go so far as to say we don't _need_ text lables under the icons at all: the existing tooltips are sufficient (the icons themselves should communicate their purpose; better not to rely on linguistic interface).

Another issue with the inital window state is that it does not consider content. It doesn't have to show all icons at once, but it shouldn't cut off the lower half of a row of icons. If we stick with this format, it should clamp to a number of whole rows (four or five, preferrably five).

Here's what it looks like [maximized](https://i.postimg.cc/wBH72sLF/switchboard-elementarydark-maximized.png). There's a lot of empty space; it would look better if the icons distributed and scaled to fill it evenly.

**Comment by danirabbit** [comment]
White space is not wasted space :)

The issue with the view being clipped with large fonts is fixed in master. It should be released soon.

Admittedly I don't think anyone is testing Switchboard maximized on a large display. There are things that could be done to make it nicer but like you pointed out most of that is things that need to be done in the plug itself. 

I just submitted this branch to address some issues: https://github.com/elementary/switchboard/pull/138

**Comment by quequotion** [comment]
Excellent! Thanks for hearing me out on this.

Next, if only it were as easy to make the icons themselves scale with the window size. I looked into this a bit last night, and found basically two ways to go: actually scaling some SVG icons (probably using Cairo; would require significant rewriting; beautiful), or swapping them with different sizes of bitmapped icons (can probably be done without new dependencies and requires somewhat less rewriting, but clunky) .

I also noticed the icons are `DND` (32px); I think `DIALOG` (48px) would be a better staring point, or a minimal size to scale up to when switchboard is maximized--unless there's a design conflict (supporting low resolution screens or matching size with something else?).

**Comment by danirabbit** [comment]
Some of these icons don't have 48px variants, but it also looks pretty comical at the default size:

![Screenshot from 2020-01-14 16 54 04@2x](https://user-images.githubusercontent.com/7277719/72395491-8170ef80-36ee-11ea-8076-ac5e47d13436.png)

I can see how this looks more natural at larger sizes though.

We could listen for the window size in `main_window.configure_event`, but the problem from there becomes getting a property change from there all the way down to each `CategoryIcon`. It's doable, but not exactly trivial.

I would recommend filing a separate issue report about this and maybe someone wants to provide a branch

**Comment by quequotion** [comment]
>it also looks pretty comical at the default size

Idk, I like it. Maybe I have a fetish for [large icons](https://www.reddit.com/r/elementaryos/comments/ekwlg9/set_everything_to_largest_so_much_better/?utm_medium=android_app&utm_source=share). 

>Some of these icons don't have 48px variants
>I would recommend filing a separate issue report about this and maybe someone wants to provide a branch

Indeed; I was expecting the scope of this to expand.

Sounds like we'll need a few more threads:

* [One to make some new 48px icons.](https://github.com/elementary/icons/issues/876) (Completed!) 
* [One to patch in scaling the icons in switchboard](https://github.com/elementary/switchboard/issues/142). (I will give a hack at it if no one else gets around to it first)
* Several to fix layouts of individual plugs. (need to research the state of each plug first)

**Comment by mrgaturus** [comment]
i noticed on macOS that you can't maximize settings window and the window resizes according the view. Compared to switchboard, maximizing it leads to a lot of wasted space on any view

**Comment by mrgaturus** [comment]
GTK4 has animations when resizing it's popovers and these are real X11/Wayland windows (as popups)
